From 4fafd4f2c8ee7f644ce266405e2464b11d89e6eb Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sun, 24 Jul 2011 18:19:40 +0200
Subject: [PATCH] [warning-fix][implicit-conversion-from-wider-type]

Changed some of the interface types (inc files) related to time_t
to correct some implicit casts.
---
 inc/Dynamic_FuzzyDefinitionsDictionary.hxx  |    6 +++++-
 inc/Dynamic_MethodDefinitionsDictionary.hxx |    6 +++++-
 inc/Materials_MaterialsDictionary.hxx       |    6 +++++-
 inc/Units_Lexicon.hxx                       |    2 +-
 inc/Units_UnitsLexicon.hxx                  |    2 +-
 5 files changed, 17 insertions(+), 5 deletions(-)

From bbb77dfcd85c5ae0163477769bbbc7681a4e0eca Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 5 Aug 2011 09:02:08 +0200
Subject: [PATCH] On Windows, use stat instead of _stat to use time_t type on
 32 and 64 bits

In Standard_Time.hxx, define Standard_Time via a typedef instead of a #define,
like other Standard_* definitions in Standard_Typedef.hxx.
---
 inc/Standard_Time.hxx                              |    9 +++++----
 src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cxx |    5 -----
 .../Dynamic_MethodDefinitionsDictionary.cxx        |    4 ----
 src/Materials/Materials_MaterialsDictionary.cxx    |    3 ---
 4 files changed, 5 insertions(+), 16 deletions(-)

From 0fd1dbf7ae818fcf2c1b79831ab597e8fb0be592 Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Fri, 13 Jan 2012 11:44:47 +0200
Subject: [PATCH 1/2] Added Standard_Time cdl based class.

In order to support oce changes in cdl defined classes.
Fix for issue #223.
---
 inc/Standard_Time.hxx     |   14 ++++++++++++++
 src/Standard/FILES        |    1 +
 src/Standard/Standard.cdl |    1 +
 3 files changed, 16 insertions(+), 0 deletions(-)
 create mode 100644 inc/Standard_Time.hxx

From 6763e93005f475b9ed3614542c4ce059b42a6e47 Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Fri, 13 Jan 2012 11:45:55 +0200
Subject: [PATCH 2/2] Modified cdl classes to use Standard_Time variable

This commit makes oce changes portable to wok system.Also solves
an issue ( #223 ) when building oce using non wok build environment.
---
 inc/Dynamic_FuzzyDefinitionsDictionary.hxx         |   13 ++++++-------
 inc/Dynamic_MethodDefinitionsDictionary.hxx        |   13 ++++++-------
 inc/Materials_MaterialsDictionary.hxx              |   13 ++++++-------
 inc/Units_Lexicon.hxx                              |    6 +++---
 inc/Units_UnitsLexicon.hxx                         |    6 +++---
 src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cdl |    2 +-
 .../Dynamic_MethodDefinitionsDictionary.cdl        |    2 +-
 src/Materials/Materials_MaterialsDictionary.cdl    |    2 +-
 src/Units/Units_Lexicon.cdl                        |    2 +-
 src/Units/Units_UnitsLexicon.cdl                   |    2 +-
 10 files changed, 29 insertions(+), 32 deletions(-)

Index: oce/inc/Dynamic_FuzzyDefinitionsDictionary.hxx
===================================================================
--- oce.orig/inc/Dynamic_FuzzyDefinitionsDictionary.hxx
+++ oce/inc/Dynamic_FuzzyDefinitionsDictionary.hxx
@@ -19,8 +19,8 @@
 #ifndef _Handle_TCollection_HAsciiString_HeaderFile
 #include <Handle_TCollection_HAsciiString.hxx>
 #endif
-#ifndef _Standard_Integer_HeaderFile
-#include <Standard_Integer.hxx>
+#ifndef _Standard_Time_HeaderFile
+#include <Standard_Time.hxx>
 #endif
 #ifndef _Handle_Dynamic_SequenceOfFuzzyDefinitions_HeaderFile
 #include <Handle_Dynamic_SequenceOfFuzzyDefinitions.hxx>
@@ -40,6 +40,9 @@
 #ifndef _Handle_Dynamic_FuzzyClass_HeaderFile
 #include <Handle_Dynamic_FuzzyClass.hxx>
 #endif
+#ifndef _Standard_Integer_HeaderFile
+#include <Standard_Integer.hxx>
+#endif
 #ifndef _Standard_OStream_HeaderFile
 #include <Standard_OStream.hxx>
 #endif
@@ -104,7 +107,7 @@ private:
 
 
 Handle_TCollection_HAsciiString thefilename;
-Standard_Integer thetime;
+Standard_Time thetime;
 Handle_Dynamic_SequenceOfFuzzyDefinitions thesequenceoffuzzydefinitions;
 
 
Index: oce/inc/Dynamic_MethodDefinitionsDictionary.hxx
===================================================================
--- oce.orig/inc/Dynamic_MethodDefinitionsDictionary.hxx
+++ oce/inc/Dynamic_MethodDefinitionsDictionary.hxx
@@ -19,8 +19,8 @@
 #ifndef _Handle_TCollection_HAsciiString_HeaderFile
 #include <Handle_TCollection_HAsciiString.hxx>
 #endif
-#ifndef _Standard_Integer_HeaderFile
-#include <Standard_Integer.hxx>
+#ifndef _Standard_Time_HeaderFile
+#include <Standard_Time.hxx>
 #endif
 #ifndef _Handle_Dynamic_SequenceOfMethodDefinitions_HeaderFile
 #include <Handle_Dynamic_SequenceOfMethodDefinitions.hxx>
@@ -40,6 +40,9 @@
 #ifndef _Handle_Dynamic_Parameter_HeaderFile
 #include <Handle_Dynamic_Parameter.hxx>
 #endif
+#ifndef _Standard_Integer_HeaderFile
+#include <Standard_Integer.hxx>
+#endif
 #ifndef _Standard_OStream_HeaderFile
 #include <Standard_OStream.hxx>
 #endif
@@ -108,7 +111,7 @@ private:
 
 
 Handle_TCollection_HAsciiString thefilename;
-Standard_Integer thetime;
+Standard_Time thetime;
 Handle_Dynamic_SequenceOfMethodDefinitions thesequenceofmethoddefinitions;
 
 
Index: oce/inc/Materials_MaterialsDictionary.hxx
===================================================================
--- oce.orig/inc/Materials_MaterialsDictionary.hxx
+++ oce/inc/Materials_MaterialsDictionary.hxx
@@ -19,8 +19,8 @@
 #ifndef _Handle_TCollection_HAsciiString_HeaderFile
 #include <Handle_TCollection_HAsciiString.hxx>
 #endif
-#ifndef _Standard_Integer_HeaderFile
-#include <Standard_Integer.hxx>
+#ifndef _Standard_Time_HeaderFile
+#include <Standard_Time.hxx>
 #endif
 #ifndef _Handle_Materials_MaterialsSequence_HeaderFile
 #include <Handle_Materials_MaterialsSequence.hxx>
@@ -37,6 +37,9 @@
 #ifndef _Standard_Boolean_HeaderFile
 #include <Standard_Boolean.hxx>
 #endif
+#ifndef _Standard_Integer_HeaderFile
+#include <Standard_Integer.hxx>
+#endif
 #ifndef _Standard_OStream_HeaderFile
 #include <Standard_OStream.hxx>
 #endif
@@ -89,7 +92,7 @@ private:
 
 
 Handle_TCollection_HAsciiString thefilename;
-Standard_Integer thetime;
+Standard_Time thetime;
 Handle_Materials_MaterialsSequence thematerialssequence;
 
 
Index: oce/inc/Standard_Time.hxx
===================================================================
--- /dev/null
+++ oce/inc/Standard_Time.hxx
@@ -0,0 +1,15 @@
+#ifndef _Standard_Time_HeaderFile
+#define _Standard_Time_HeaderFile
+
+// File    : Standard_Time.hxx
+// Created : Fri Jan 13 2012 for OpenCascade Community edition
+// Author  : Fotis Sioutis <sfotis at gmail dot com>
+// Purpose : This file is intented to provide inclusion of 'time' 
+//           variable in a CDL consistent way.
+
+/* time.h is well standardised across various platforms */
+#include <time.h>
+
+typedef time_t Standard_Time;
+
+#endif
Index: oce/inc/Units_Lexicon.hxx
===================================================================
--- oce.orig/inc/Units_Lexicon.hxx
+++ oce/inc/Units_Lexicon.hxx
@@ -19,8 +19,8 @@
 #ifndef _Handle_TCollection_HAsciiString_HeaderFile
 #include <Handle_TCollection_HAsciiString.hxx>
 #endif
-#ifndef _Standard_Integer_HeaderFile
-#include <Standard_Integer.hxx>
+#ifndef _Standard_Time_HeaderFile
+#include <Standard_Time.hxx>
 #endif
 #ifndef _Handle_Units_TokensSequence_HeaderFile
 #include <Handle_Units_TokensSequence.hxx>
@@ -85,7 +85,7 @@ private:
 
 
 Handle_TCollection_HAsciiString thefilename;
-Standard_Integer thetime;
+Standard_Time thetime;
 Handle_Units_TokensSequence thesequenceoftokens;
 
 
Index: oce/inc/Units_UnitsLexicon.hxx
===================================================================
--- oce.orig/inc/Units_UnitsLexicon.hxx
+++ oce/inc/Units_UnitsLexicon.hxx
@@ -19,8 +19,8 @@
 #ifndef _Handle_TCollection_HAsciiString_HeaderFile
 #include <Handle_TCollection_HAsciiString.hxx>
 #endif
-#ifndef _Standard_Integer_HeaderFile
-#include <Standard_Integer.hxx>
+#ifndef _Standard_Time_HeaderFile
+#include <Standard_Time.hxx>
 #endif
 #ifndef _Units_Lexicon_HeaderFile
 #include <Units_Lexicon.hxx>
@@ -72,7 +72,7 @@ private:
 
 
 Handle_TCollection_HAsciiString thefilename;
-Standard_Integer thetime;
+Standard_Time thetime;
 
 
 };
Index: oce/src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cdl
===================================================================
--- oce.orig/src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cdl
+++ oce/src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cdl
@@ -128,7 +128,7 @@ is
 fields
 
     thefilename                   : HAsciiString               from TCollection;
-    thetime                       : Integer from Standard                                ;
+    thetime                       : Time from Standard                                ;
     thesequenceoffuzzydefinitions : SequenceOfFuzzyDefinitions from Dynamic    ;
 
 end FuzzyDefinitionsDictionary;
Index: oce/src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cxx
===================================================================
--- oce.orig/src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cxx
+++ oce/src/Dynamic/Dynamic_FuzzyDefinitionsDictionary.cxx
@@ -50,11 +50,6 @@
 # include <strings.h>
 #endif
 
-#ifdef WNT
-//#define strcasecmp _stricoll
-#define stat _stat
-#endif
-
 //=======================================================================
 //function : Dynamic_FuzzyDefinitionsDictionary
 //purpose  : 
Index: oce/src/Dynamic/Dynamic_MethodDefinitionsDictionary.cdl
===================================================================
--- oce.orig/src/Dynamic/Dynamic_MethodDefinitionsDictionary.cdl
+++ oce/src/Dynamic/Dynamic_MethodDefinitionsDictionary.cdl
@@ -136,7 +136,7 @@ is
 fields
 
     thefilename                    : HAsciiString                from TCollection;
-    thetime                        : Integer from Standard                                ;
+    thetime                        : Time from Standard                                ;
     thesequenceofmethoddefinitions : SequenceOfMethodDefinitions from Dynamic    ;
 
 end MethodDefinitionsDictionary;
Index: oce/src/Dynamic/Dynamic_MethodDefinitionsDictionary.cxx
===================================================================
--- oce.orig/src/Dynamic/Dynamic_MethodDefinitionsDictionary.cxx
+++ oce/src/Dynamic/Dynamic_MethodDefinitionsDictionary.cxx
@@ -51,10 +51,6 @@
 #ifdef HAVE_STRINGS_H
 # include <strings.h>
 #endif
-#ifdef WNT
-#define stat _stat
-//#define strcasecmp _stricoll
-#endif
 
 //=======================================================================
 //function : Dynamic_MethodDefinitionsDictionary
Index: oce/src/Materials/Materials_MaterialsDictionary.cdl
===================================================================
--- oce.orig/src/Materials/Materials_MaterialsDictionary.cdl
+++ oce/src/Materials/Materials_MaterialsDictionary.cdl
@@ -85,7 +85,7 @@ is
 fields
 
     thefilename          : HAsciiString from TCollection;
-    thetime              : Integer from Standard;
+    thetime              : Time from Standard;
     thematerialssequence : MaterialsSequence from Materials;
 
 end MaterialsDictionary;
Index: oce/src/Materials/Materials_MaterialsDictionary.cxx
===================================================================
--- oce.orig/src/Materials/Materials_MaterialsDictionary.cxx
+++ oce/src/Materials/Materials_MaterialsDictionary.cxx
@@ -48,9 +48,6 @@
 #include <Quantity_Color.hxx>
 #include <TCollection_AsciiString.hxx>
 
-#ifdef WNT
-#define stat _stat
-#endif
 //#define strcasecmp _stricoll
 #include <stdio.h>
 //#endif
Index: oce/src/Standard/FILES
===================================================================
--- oce.orig/src/Standard/FILES
+++ oce/src/Standard/FILES
@@ -73,4 +73,5 @@ Standard_MMgrTBBalloc.cxx
 Standard_MMgrTBBalloc.hxx
 Standard_Assert.hxx
 Standard_DefineAlloc.hxx
+Standard_Time.hxx
 Standard_StdAllocator.hxx
Index: oce/src/Standard/Standard.cdl
===================================================================
--- oce.orig/src/Standard/Standard.cdl
+++ oce/src/Standard/Standard.cdl
@@ -66,6 +66,7 @@ is
     primitive PByte;
     primitive PCharacter;
     primitive PExtCharacter;
+	primitive Time;
 
     deferred class ErrorHandlerCallback;
     class ErrorHandler;
Index: oce/src/Units/Units_Lexicon.cdl
===================================================================
--- oce.orig/src/Units/Units_Lexicon.cdl
+++ oce/src/Units/Units_Lexicon.cdl
@@ -105,7 +105,7 @@ is
 fields
 
     thefilename         : HAsciiString from TCollection;
-    thetime             : Integer;
+    thetime             : Time;
     thesequenceoftokens : TokensSequence from Units;
 
 end Lexicon;
Index: oce/src/Units/Units_UnitsLexicon.cdl
===================================================================
--- oce.orig/src/Units/Units_UnitsLexicon.cdl
+++ oce/src/Units/Units_UnitsLexicon.cdl
@@ -85,6 +85,6 @@ is
 fields
 
     thefilename     : HAsciiString from TCollection;
-    thetime         : Integer;
+    thetime         : Time;
 
 end UnitsLexicon;
